core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -217,8 +217,13 @@ private Path unzip(Path zip, Path pluginsDir) throws IOException {
         try (ZipInputStream zipInput = new ZipInputStream(Files.newInputStream(zip))) {
             ZipEntry entry;
             byte[] buffer = new byte[8192];
+            String root = "elasticsearch" + pluginsDir.getFileSystem().getSeparator();
```

Comment:
zips do not use the filesystem separate, they always use `/`. I don't think this is necessary. Below we should just use `substring("elasticsearch/".length())`